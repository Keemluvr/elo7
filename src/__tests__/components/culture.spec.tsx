import { render } from "@testing-library/react";
import Culture from "@/components/culture";

describe("Culture component", () => {
  describe("Section", () => {
    test("should render the section component", () => {
      const { getByRole } = render(<Culture />);
      const sectionElement = getByRole("region", {
        name: "seção sobre a cultura da elo7",
      });
      expect(sectionElement).toBeInTheDocument();
    });

    test("should render the section with correct tag name", () => {
      const { getByRole } = render(<Culture />);
      const sectionElement = getByRole("region", {
        name: "seção sobre a cultura da elo7",
      });
      expect(sectionElement.tagName).toBe("SECTION");
    });
  });

  describe("List", () => {
    test("should render the list ", () => {
      const { getByRole } = render(<Culture />);
      const sectionElement = getByRole("region", {
        name: "seção sobre a cultura da elo7",
      });
      expect(sectionElement).toBeInTheDocument();
    });

    test("should render three CultureCard components", () => {
      const { getAllByRole } = render(<Culture />);
      const cultureCards = getAllByRole("listitem", {
        name: "card de cultura",
      });
      expect(cultureCards).toHaveLength(3);
    });
  });

  test("matches snapshot", () => {
    const { asFragment } = render(<Culture />);
    expect(asFragment()).toMatchSnapshot();
  });
});
