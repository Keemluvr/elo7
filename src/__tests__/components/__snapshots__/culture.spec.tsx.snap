// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Culture component matches snapshot 1`] = `
<DocumentFragment>
  <section
    aria-label="seção sobre a cultura da elo7"
    class="grid-container grid-item-full-width-without-margin culture-wrapper"
  >
    <ul
      aria-label="listagem dos valores da elo7"
      class="grid-item-full-width culture-content"
    >
      <li
        aria-label="card de cultura"
        class="culture-card-wrapper"
      >
        <img
          alt="ícone de uma imagem"
          class="culture-card-icon"
          data-nimg="1"
          decoding="async"
          height="136"
          loading="lazy"
          src="/_next/image?url=%2Fimages%2Fculture%2Fgallery.webp&w=384&q=100"
          srcset="/_next/image?url=%2Fimages%2Fculture%2Fgallery.webp&w=256&q=100 1x, /_next/image?url=%2Fimages%2Fculture%2Fgallery.webp&w=384&q=100 2x"
          style="color: transparent;"
          width="136"
        />
        <h2
          class="culture-card-title"
        >
          Resultados
        </h2>
        <p
          class="culture-card-description"
        >
          Donec in vestibulum elit. Aliquam finibus facilisis elit, sit amet malesuada nibh tempor sed. Aliquam consequat ultrices fringilla. Sed id quam sollicitudin mi ultricies feugiat a vel velit. Pellentesque finibus vel tortor sed hendrerit. Vestibulum eu sem sapien.
        </p>
      </li>
      <li
        aria-label="card de cultura"
        class="culture-card-wrapper"
      >
        <img
          alt="ícone de um polegar para cima"
          class="culture-card-icon"
          data-nimg="1"
          decoding="async"
          height="136"
          loading="lazy"
          src="/_next/image?url=%2Fimages%2Fculture%2Fthumbs-up.webp&w=384&q=100"
          srcset="/_next/image?url=%2Fimages%2Fculture%2Fthumbs-up.webp&w=256&q=100 1x, /_next/image?url=%2Fimages%2Fculture%2Fthumbs-up.webp&w=384&q=100 2x"
          style="color: transparent;"
          width="136"
        />
        <h2
          class="culture-card-title"
        >
          Qualidade de vida
        </h2>
        <p
          class="culture-card-description"
        >
          Donec in vestibulum elit. Aliquam finibus facilisis elit, sit amet malesuada nibh tempor sed. Aliquam consequat ultrices fringilla.
        </p>
        <a
          class="culture-card-button"
          href="https://www.elo7.com.br/sobre"
          rel="noopener noreferrer"
          role="button"
          target="_blank"
        >
          dignissim quis
        </a>
      </li>
      <li
        aria-label="card de cultura"
        class="culture-card-wrapper"
      >
        <img
          alt="ícone de um cofre"
          class="culture-card-icon"
          data-nimg="1"
          decoding="async"
          height="136"
          loading="lazy"
          src="/_next/image?url=%2Fimages%2Fculture%2Fvault.webp&w=384&q=100"
          srcset="/_next/image?url=%2Fimages%2Fculture%2Fvault.webp&w=256&q=100 1x, /_next/image?url=%2Fimages%2Fculture%2Fvault.webp&w=384&q=100 2x"
          style="color: transparent;"
          width="136"
        />
        <h2
          class="culture-card-title"
        >
          Resultados
        </h2>
        <p
          class="culture-card-description"
        >
          Donec in vestibulum elit. Aliquam finibus facilisis elit, sit amet malesuada nibh tempor sed. Aliquam consequat ultrices fringilla. Sed id quam sollicitudin mi ultricies feugiat a vel velit. Pellentesque finibus vel tortor sed hendrerit. Vestibulum eu sem sapien.
        </p>
      </li>
    </ul>
  </section>
</DocumentFragment>
`;
